sw/device/silicon_creator/lib/ownership/owner_block.c

```diff
@@ -143,6 +144,12 @@ rom_error_t owner_block_parse(const owner_block_t *block,
           return kErrorOwnershipDuplicateItem;
         config->rescue = (const owner_rescue_config_t *)item;
         break;
+      case kTlvTagIntegrationSpecificFirmwareBinding:
+        HARDENED_CHECK_EQ(tag, kTlvTagIntegrationSpecificFirmwareBinding);
+        if ((hardened_bool_t)config->isfb != kHardenedBoolFalse)
+          return kErrorOwnershipDuplicateItem;
+        config->isfb = (const owner_isfb_config_t *)item;
```

Comment:
We need to sanity check the ISFB configuration.  Notably, we need to ensure the selected flash bank & page are within the range designated as owner info pages.  There is a helper function `is_owner_page` below in this file.  It's currently tailored to the info config struct, but it could be made more generic.

Reply:
Added sanity check function. Also added unittests to the `owner_block_unittest` target. I ended up finding a bug in `opentitanlib` which has now been fixed in a separate  commit included in this PR.